x/incentives/client/cli/query.go

```diff
@@ -384,31 +386,80 @@ $ %s query incentives rewards-estimation
 				return err
 			}
 
-			lockIdStrs := strings.Split(lockIdsCombined, ",")
-			lockIds := []uint64{}
-			for _, lockIdStr := range lockIdStrs {
-				lockId, err := strconv.ParseUint(lockIdStr, 10, 64)
+			var res *lockuptypes.AccountLockedLongerDurationResponse
+			if owner != "" {
+				queryClientLockup := lockuptypes.NewQueryClient(clientCtx)
+
+				res, err = queryClientLockup.AccountLockedLongerDuration(cmd.Context(), &lockuptypes.AccountLockedLongerDurationRequest{Owner: owner, Duration: time.Millisecond})
 				if err != nil {
 					return err
 				}
-				lockIds = append(lockIds, lockId)
+			} else {
+				owner = ""
+			}
+
+			ownerLocks := []uint64{}
+
+			if res != nil {
+				for _, lockId := range res.Locks {
+					ownerLocks = append(ownerLocks, lockId.ID)
+				}
+			}
+
+			lockIdStrs := strings.Split(lockIdsCombined, ",")
+			lockIds := []uint64{}
+			// if user doesn't provide any lockIDs or a lock owner, we don't have enough information to proceed
+			if lockIdsCombined == "" && owner == "" {
+				err = fmt.Errorf("if owner flag is not set, lock IDs must be provided")
+				return err
+
+				// if user provides lockIDs, use these lockIDs in our rewards estimation
+			} else if lockIdsCombined != "" {
+				for _, lockIdStr := range lockIdStrs {
+					lockId, err := strconv.ParseUint(lockIdStr, 10, 64)
+					if err != nil {
+						return err
+					}
+					lockIds = append(lockIds, lockId)
+				}
+
+				// if no lockIDs are provided but an owner is provided, we query the rewards for all of the locks the owner has
+			} else if lockIdsCombined == "" && owner != "" {
+				lockIds = append(lockIds, ownerLocks...)
+			}
+
+			// if lockIDs are provided and an owner is provided, only query the lockIDs that are provided
+			// if a lockID was provided and it doesn't belong to the owner, return an error
+			if len(lockIds) != 0 && owner != "" {
+				for _, lockId := range lockIds {
+					validInputLockId := contains(ownerLocks, lockId)
+					if !validInputLockId {
+						return fmt.Errorf("lock-id %v does not belong to %v", lockId, owner)
+					}
+				}
 			}
 
 			endEpoch, err := cmd.Flags().GetInt64(FlagEndEpoch)
 			if err != nil {
 				return err
 			}
 
-			res, err := queryClient.RewardsEst(cmd.Context(), &types.RewardsEstRequest{
+			// TODO: Figure out why some lock ids are good and some causes "Error: rpc error: code = Unknown desc = panic message redacted to hide potentially sensitive system info: panic"
+			// since owner checks have already been completed above, we switch the owner address to a random module account address since a blank owner panics.
+			// we should find a better way to circumvent this address validity check
+			if owner == "" {
+				owner = "osmo14kjcwdwcqsujkdt8n5qwpd8x8ty2rys5rjrdjj"
+			}
+			res1, err1 := queryClient.RewardsEst(cmd.Context(), &types.RewardsEstRequest{
```

Comment:
Let's reuse `res` and `err` please

Reply:
Can you please show me how to reuse res if the prior definition was of a different type?

Reply:
If `res1` is of a different type, let's choose a more meaningful name instead such as `rewardsEstimateResult`? And for `res` something like `lockedAccountsResult`? That would help to reduce the similarities with names.

`err` might still be possible to reuse

Reply:
Sweet, sounds good! Just was making sure I wasn't missing some special way to redeclare a variable 👍